components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/ServiceBusInitializer.java

```diff
@@ -165,6 +175,20 @@ protected void activate(ComponentContext ctxt) {
 
                 synapseEnvironment.registerSynapseHandler(new SynapseExternalPropertyConfigurator());
                 synapseEnvironment.registerSynapseHandler(new ProxyLogHandler());
+
+                // Register internal transaction synapse handler
+                boolean transactionPropertyEnabled = TransactionHandlerComponent.isTransactionPropertyEnabled();
+                if (transactionPropertyEnabled) {
+                    TransactionHandler transactionHandler = new TransactionHandler();
+                    synapseEnvironment.registerSynapseHandler(transactionHandler);
+                    try {
+                        transactionHandlerComponent = new TransactionHandlerComponent();
+                        transactionHandlerComponent.start(dataSourceService);
+                    } catch (DataSourceException | TransactionException e) {
+                        log.warn("Error in initializing Transaction Service component: " + e.getMessage(), e);
```

Comment:
Would the level ERROR be better since we are logging an error anyway. Also , we don't need to log e.getMessage() it'll be logged when you have the exception as a parameter

Reply:
Also, now that anybody who does not have a datasource configured will see this message, I think it should be a bit more descriptive. "Error in initializing the Transaction component. The number of transactions will not be recorded. You can ignore this error if you don't want the server to keep track of the number of transactions processed"

Reply:
+1